ui/app/components/clients/attribution.js

```diff
@@ -86,37 +85,50 @@ export default class Attribution extends Component {
     }
   }
 
-  // TODO CMB update with proper data format when we have
   get getCsvData() {
-    let results = '',
-      data,
-      fields;
-
-    // TODO CMB will CSV for namespaces include mounts?
-    fields = [`${this.attributionBreakdown}`, 'Active clients', 'Unique entities', 'Non-entity tokens'];
-
-    results = fields.join(',') + '\n';
-    data.forEach(function (item) {
-      let path = item.label !== '' ? item.label : 'root',
-        total = item.total,
-        unique = item.entity_clients,
-        non_entity = item.non_entity_clients;
+    let csvData = [],
+      graphData = this.totalClientsData,
+      csvHeader = [
+        `Namespace path`,
+        'Authentication method',
+        'Total clients',
+        'Entity clients',
+        'Non-entity clients',
+      ];
 
-      results += path + ',' + total + ',' + unique + ',' + non_entity + '\n';
-    });
-    return results;
+    // each array will be a row in the csv file
+    if (this.isSingleNamespace) {
+      graphData.forEach((mount) => {
+        csvData.push(['', mount.label, mount.clients, mount.entity_clients, mount.non_entity_clients]);
+      });
+      csvData.forEach((d) => (d[0] = this.args.selectedNamespace));
+    } else {
+      graphData.forEach((ns) => {
+        csvData.push([ns.label, '', ns.clients, ns.entity_clients, ns.non_entity_clients]);
+        if (ns.mounts) {
+          ns.mounts.forEach((m) => {
+            csvData.push([ns.label, m.label, m.clients, m.entity_clients, m.non_entity_clients]);
+          });
+        }
+      });
+    }
+    csvData.unshift(csvHeader);
+    // make each nested array a comma separated string, join each array in csvData with line break (\n)
+    return csvData.map((d) => d.join()).join('\n');
   }
-  // TODO CMB - confirm with design file name structure
+
   get getCsvFileName() {
-    let activityDateRange = `${this.args.startTimeDisplay} - ${this.args.endTimeDisplay}`;
-    return activityDateRange
-      ? `clients-by-${this.attributionBreakdown}-${activityDateRange}`
-      : `clients-by-${this.attributionBreakdown}-${new Date()}`;
+    let endRange = this.isDateRange ? `-${this.args.endTimeDisplay}` : '';
```

Comment:
If there's no end time, maybe we put today's date?

Reply:
There should only be no `endRange` if the user is viewing the current month tab so only one month is displaying